pansat/download/accounts.py

```diff
@@ -2,54 +2,192 @@
-   directory. This is a security issue and fixed as soon as possible.
-
+The ``accounts`` sub-module handles login data for different data portals.
+The login data is stored in encrypted format in a configuration
+file ``identities.json`` in the user's home directory tree.
+
+Upon first usage the identities file is setup with a custom user password.
+This password is used to encrypt all data passwords passwords that is
+ subsequently added by the user. The Fernet method is used to en- and
+decrypt the passwords. All password hashing is performed using random
+salt.
 """
-from appdirs import user_config_dir
+import base64
 import getpass
 import json
-import keyring
 import os
 from pathlib import Path
 
-identities = {}
+from appdirs import user_config_dir
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+
+_IDENTITIES = {}
 """
-Dictionaries containing provider names and corresponding user names.
+Dictionary containing provider names and corresponding user names.
 """
 
 # The directory containing the configuration file.
-_app_dir = Path(user_config_dir("pansat", "pansat"))
-_app_dir.mkdir(parents=True, exist_ok=True)
+_APP_DIR = Path(user_config_dir("pansat", "pansat"))
+_APP_DIR.mkdir(parents=True, exist_ok=True)
 
 # The path to the configuration file.
-_identity_file = Path(_app_dir)
-_identity_file /= Path("identities.json")
+_IDENTITY_FILE = Path(_APP_DIR)
+_IDENTITY_FILE /= Path("identities.json")
+
+_PANSAT_SECRET = None
 
 ###############################################################################
 # Handling identities
 ###############################################################################
 
 
+def get_password(check=False):
+    """
+    Check if password is provided as ``PANSAT_PASSWORD`` variable.
+    If this is not the case query user to input password.
+
+    Params:
+        check(``bool``): Whether or user should insert the password twice
+            to avoid spelling errors.
+
+    Return:
+        ``str`` containing the password
+    """
+    try:
+        password = os.environ["PANSAT_PASSWORD"]
+    except KeyError:
+        print("Please enter your pansat user password:")
+        password = getpass.getpass()
+
+        if check:
+            print("Please repeat the password:")
+            password_re = getpass.getpass()
+            if password != password_re:
+                print("The two passwords don't match!")
+                get_password(check=check)
+
+    return password
+
+
+def hash_password(password, salt):
+    """
+    Computes the hash of a password.
+
+    Args:
+        password(``str``): The password
+        slat(``bytes``): The salt to use to compute the hash.
+
+    Return:
+        The base64 encoded hash of the password.
+    """
+    password = password.encode()
+    kdf = PBKDF2HMAC(algorithm=hashes.SHA256(), length=32, salt=salt, iterations=100000)
```

Comment:
This seems to be a version issue. In newer versions of cryptography>=3.1, ``backend`` is a keyword argument, which
defaults to ``None``. So I suggest to add this to ``install_requirements`` instead, since I am unsure whether we can rely on
the ``default_backend()`` solution.